src/stores/batch-store.js

```diff
@@ -1,11 +1,38 @@
 import Database from './database.js';
-import StoreBase from './store-base.js';
 
-class BatchStore extends StoreBase {
+class BatchStore {
   static get inject() { return [Database]; }
 
   constructor(db) {
-    super(db.batches);
+    this.collection = db.batches;
+  }
+
+  getBatchById(batchId, { session } = {}) {
+    return this.collection.findOne({ _id: batchId }, { session });
+  }
+
+  getBatchesByType(batchType, { session } = {}) {
+    return this.collection.find({ batchType }, { session }).toArray();
+  }
+
+  getUncompleteBatchByType(batchType, { session } = {}) {
+    return this.collection.findOne({ batchType, completedOn: null }, { session });
+  }
+
+  getUncompletedBatchByTypeAndHost({ batchType, hostName }, { session } = {}) {
+    return this.collection.findOne({ batchType, 'batchParams.hostName': hostName, 'completedOn': null }, { session });
+  }
+
+  getUncompletedBatch({ session } = {}) {
+    return this.collection.findOne({ completedOn: null }, { session });
+  }
+
+  addBatch(batch, { session } = {}) {
```

Comment:
Or should we go with `insertBatch` in order to stay closer to the MongoDB terminology? I know that we translate `get` into `find` and `save` into `upsert`, but `add` sounds very weak to me. Alternatively. I could imaging `createBatch` (like the `C` in `CRUD`). Just a thought, in the end we should just stick to one terminology whichever it is.

Reply:
`createBatch` actually would be better, will change.